tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/SMTEncodings.scala

```diff
@@ -0,0 +1,6 @@
+package at.forsyte.apalache.tla.bmcmt
+
+object SMTEncodings extends Enumeration {
+  type SMTEncoding = Value
+  val oopsla19Encoding, arraysEncoding, invalidEncoding = Value
+}
```

Comment:
Sorry for the churn here, but your remark about the lack of support for enums in the CLI library led me to look up info about enums in Scala, only to find that, for enums:

> There’s no exhaustive matching check during compile

This defeats about 50% of the advantage from using some kind of disjoint union here. This is due to lack of clarity in my own expressions, tho I also blame it partially on Scala having multiple,  variously defective, ways of achieving the same thing.

E.g., in the [scalabook docs on "Case Objecs"](https://docs.scala-lang.org/overviews/scala-book/case-objects.html) they write:

> Creating enumerations with case objects
>
> As we showed earlier in this book, you create enumerations in Scala like this:
>
> ```
> sealed trait Topping
> case object Cheese extends Topping
> case object Pepperoni extends Topping
> case object Sausage extends Topping
> case object Mushrooms extends Topping
> case object Onions extends Topping
> ```

That is, according to this resource, "you create enumerations" not by using `Enumeration` (as one might very well think!) but by defining case classes/case objects :/

So, again, sorry for the churn, but what we really want here is a (sealed) case object (or case class, if we end up needing to carry around any additional data).

This will give us 

- exhaustiveness checking, meaning there shouldn't be any need for the `invalidEncoding` value any more
- ability to parse the CLI argument directly into the proper type, so we don't have to do the conversion twice.

Reply:
No need to apologize, it is a good thing that you caught this! I will try to make the change to case objects as suggested. Thanks!